### ISSUE TITLE
Figure out how to handle feedback submissions consistently

### ISSUE THREAD

**Issue by lmorchard** [mozilla/testpilot#430]
The main website has [feedback forms](https://github.com/mozilla/testpilot/blob/master/testpilot/frontend/static-src/app/views/feedback-view.js#L81) attached to the enable/disable buttons. These submit to [`/api/feedback`](https://github.com/mozilla/testpilot/blob/master/testpilot/experiments/views.py#L89), which stores the responses as database rows.

The addon (as of #400) has a feedback survey door hanger that [submits feedback as a metrics ping](https://github.com/mozilla/testpilot/blob/master/addon/index.js#L162).

I think the site and the addon should both do the same thing with respect to feedback. 

So, the question is: Should feedback go into the database (and we change the addon)? Or should it be a form of metric ping (and we change the site)? Depends on what we want to do with feedback in terms of review & analysis (which we should also figure out).

**Comment by clouserw** [comment]
I've been telling everyone (privacy, legal, metrics, etc.) that we're storing it in the database, but Javaun has more ideas and will write below

**Comment by javaun** [comment]
Matt Grimes has an ask here that we use SurveyGizmo forms whenever we ask for Feedback. (general opt-in, leaving an experiment, leaving test pilot entirely).

This is opening a tab and handing off to a third-party survey that UAG administers. The rationale is: when someone is talking to us, this is our best chance for directional product feedback, and currently we have the same 4 buttons and open text field for all feedback forms. We have experience in gathering feedback and typically this kind of feedback isn't hugely helpful. The suggestion is that UAG sets up a separate contextual survey for each UI instance / feature. UAG would write the surveys. This would mean we store no feedback and it all goes into our SurveyGizmo instance.

**Comment by lmorchard** [comment]
Some questions / thoughts on this:
- We have feedback forms 1) on the site content and 2) in the chrome door hanger. Does this mean those become just buttons that launch a new tab with a SurveyGizmo URL?
- How many surveys per experiment? Currently, we have spots to ask 1) at points during participation and 2) when the user disables the experiment. Should those be different survey URLs or the same? Or, will we want to support a totally different scheme/schedule of feedback gathering?
- I think feedback when leaving Test Pilot entirely is a new feature - I don't remember seeing an issue that captures that work.
- How does this flow work with disabling an experiment? Probably easiest to just remove the experiment add-on and open the new tab in parallel. ([Sounds like a "thank you" is in order, at least](https://github.com/mozilla/testpilot/issues/536))
- What's the process for getting the survey(s) set up for an experiment? Do we manage that for experimenters? Or do experimenters go directly to UAG and they hand off the survey URLs to us for entry into the site admin?

**Comment by johngruen** [comment]
~~Would have been nice to have been flagged on this before i spent time cleaning up form validation and modal UI for several hours. Also because it affects UX pretty seriously in general.~~

Nevermind...just read time stamps. Will yell at Matt Grimes tomorrow :)

**Comment by javaun** [comment]
Sorry John. I heard this today in my 1:1, just prior to triage. I don't have more information than I shared here.

**Comment by ghost** [comment]
We discussed this in the meeting today and concluded we should be using surveygizmo.  John is going to run through and answer the above questions and get the survey info and then hand off to development.  Thanks!

**Comment by johngruen** [comment]
```
Some questions / thoughts on this:

We have feedback forms 1) on the site content and 2) in the chrome door hanger. Does this mean those  become just buttons that launch a new tab with a SurveyGizmo URL?
```

New Doorhanger UI backs out a lot of this functionality. the doorhanger now shows only links to each experiment, if they are active and a link ot the experiments home page:
https://people.mozilla.org/~jgruen/test-pilot/main-doorhanger.html

Surveys are now accessed only through individual experiment pages. @bryanbell and I are working on a button design that experiment devs could add to their experiments to link to the relevant survey gizmo page but that wouldn't have any dependencies on the Test Pilot service.

```
How many surveys per experiment? Currently, we have spots to ask 1) at points during participation and 2) when the user disables the experiment. Should those be different survey URLs or the same? Or, will we want to support a totally different scheme/schedule of feedback gathering?
```

~~Two surveys per experiment (feedback and disable)~~ plus an additional survey attached to the retire flow: 
https://people.mozilla.org/~jgruen/test-pilot/retire.html

EDIT: Matt Grimes is commenting below, but i think we will get 1 URL from survey gizmo and then pass it back with different query params based on what type of feedback initiated it. Matt will comment below.

```
I think feedback when leaving Test Pilot entirely is a new feature - I don't remember seeing an issue that captures that work.
```

It's new and probably doesn't block launch

```
How does this flow work with disabling an experiment? Probably easiest to just remove the experiment add-on and open the new tab in parallel. (Sounds like a "thank you" is in order, at least)
```

Yep

```
What's the process for getting the survey(s) set up for an experiment? Do we manage that for experimenters? Or do experimenters go directly to UAG and they hand off the survey URLs to us for entry into the site admin?
```

We're using SurveyGizmo so we can do the latter option. UAG will be responsible for crafting surveys and providing the TxP team with URLs.

**Comment by MattGrimes** [comment]
Hey folks. John asked me to chime in here. My current understanding of the feedback needs are as follows.

These are the interaction points for individual experiments:
1. Unsolicited feedback from the experiment page
2. Prompt for feedback from users opting out of a study
3. Prompt for feedback from a random population of users AFTER giving the experiment a star rating

These can all be handled with a single survey. We just need to pass the appropriate variables in so that we can sort the data accordingly. 1 and 2 will get you the issues people are the most passionate about, but that data won't generalize. 3 will get you the most generalizable feedback as the sample is selected at random. It won't have the same negative biases as the other 2 sources so it will be much more representative. 

These variables not only provide better analysis, but allow us to change the questions as needed. We'll want to ask some general usability questions to all groups. The opt-out group will also get additional questions specifically around the reasons they left the experiment.

You can read the documentation on url variables here: http://help.surveygizmo.com/help/article/link/url-merge-codes

Proposed variables:
- Source: 1,2,3 from above
- Star Rating: This will only be needed from the random sample cohort
- Other experiments: We'll probably want to know if you are using other experiments as they may cause conflicts
- Experiment Branches: If you are doing a/b/c experiments we'll need to know which branch you belong to

I think that should cover it. @rayborn will be handling the implementation from our side and is available to answer questions as well.

**Comment by johngruen** [comment]
@MattGrimes In addition there will be one more survey which is the retire flow survey. this is for test pilot itself and should only be a one-off.

**Comment by MattGrimes** [comment]
Correct. Sorry, I forgot to mention that.

**Comment by johngruen** [comment]
@lmorchard @clouserw @meandavejustice @ckprice I've broken this down into four implementation bugs with the `SurveyGizmo implementation` tag.
- [ ] #568 Implement new SurveyGizmo-based experiment feedback flow
- [ ] #566 Implement random survey behavior
- [ ] #536 Implement quit experiment/thank you flow
- [ ] #499 Implement 'Retire' flow to allow users to uninstall Test Pilot

**Comment by lmorchard** [comment]
Going to close this issue, because I think we've figured out a plan to handle feedback submissions consistently and @johngruen helpfully broke it down into tasks.
